From 2e53a1ed6b35131fce63d0fca5e01948d6bc1b5f Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Fri, 6 Jul 2018 22:34:51 +0800
Subject: [PATCH] ARM: dts: rk3308-voice-module: enable
 'rockchip,en-always-grps' with AMIC boards

If the VAD which using ACODEC and AMIC is active,
we need to keep ACODEC ADCs enabled during suspend,
otherwise, the system can not be waken up.

Change-Id: I0e6d07608f5ec514f94013a255d9061a138fbee2
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 .../dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts   | 5 +++++
 arch/arm/boot/dts/rk3308-voice-module-board-v10-aarch32.dts  | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/arch/arm/boot/dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts b/arch/arm/boot/dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts
index a05ecdb37d72..5dcc61371bc4 100644
--- a/arch/arm/boot/dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts
+++ b/arch/arm/boot/dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts
@@ -13,6 +13,11 @@
 	compatible = "rockchip,rk3308-voice-module-amic-mainboard-v10-aarch32", "rockchip,rk3308";
 };
 
+&acodec {
+	/* The grp1/2 (ADC3-6) keeps enabled always for VAD */
+	rockchip,en-always-grps = <1 2>;
+};
+
 &is31fl3236 {
 	reg = <0x3f>;
 };
diff --git a/arch/arm/boot/dts/rk3308-voice-module-board-v10-aarch32.dts b/arch/arm/boot/dts/rk3308-voice-module-board-v10-aarch32.dts
index 2efd960abf4b..4b96198a4615 100644
--- a/arch/arm/boot/dts/rk3308-voice-module-board-v10-aarch32.dts
+++ b/arch/arm/boot/dts/rk3308-voice-module-board-v10-aarch32.dts
@@ -27,4 +27,6 @@
 	rockchip,no-deep-low-power;
 	/* ADC5-6 for 2ch capture, loopback to ADC3-4 */
 	rockchip,adc-grps-route = <2 1 3 0>;
+	/* The grp2 (ADC5-6) keeps enabled always for VAD */
+	rockchip,en-always-grps = <2>;
 };
-- 
2.35.3

